### ISSUE TITLE
Netlify and Vercel lambda functions expect CJS, not ESM

### ISSUE THREAD

**Issue by GrygrFlzr** [sveltejs/kit#400]
**Describe the bug**
For both Netlify and Vercel, the functions are expected to be in CJS format.

**Logs**
<details><summary>Netlify</summary>

```json
{
    "errorType": "Runtime.UserCodeSyntaxError",
    "errorMessage": "SyntaxError: Cannot use import statement outside a module",
    "trace": [
        "Runtime.UserCodeSyntaxError: SyntaxError: Cannot use import statement outside a module",
        "    at _loadUserApp (/var/runtime/UserFunction.js:98:13)",
        "    at Object.module.exports.load (/var/runtime/UserFunction.js:140:17)",
        "    at Object.<anonymous> (/var/runtime/index.js:43:30)",
        "    at Module._compile (internal/modules/cjs/loader.js:999:30)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)",
        "    at Module.load (internal/modules/cjs/loader.js:863:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:708:14)",
        "    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)",
        "    at internal/main/run_main_module.js:17:47"
    ]
}
```
</details>

<details><summary>Vercel</summary>

```
2021-02-21T09:39:07.112Z	undefined	ERROR	/var/task/index.js:1
import { parse, URLSearchParams } from 'url';
^^^^^^
SyntaxError: Cannot use import statement outside a module
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at /var/task/___now_launcher.js:26:28
    at Object.<anonymous> (/var/task/___now_launcher.js:74:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
RequestId: 4f971773-0d4d-4914-90d4-eb2f6706c2c8 Error: Runtime exited with error: exit status 1
Runtime.ExitError
```
</details>

**To Reproduce**
1. Use any of the example apps in `kit/examples`
2. Build and then adapt with the Netlify adapter
3. Initialize the project on Netlify
4. Build will succeed, but all pages will error as the function cannot run

Repeat above with Vercel.

**Expected behavior**
No import issues.

**Information about your SvelteKit Installation:**

```
  System:
    OS: Windows 10 10.0.19042
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 8.05 GB / 31.95 GB
  Binaries:
    Node: 14.15.3 - C:\Program Files\nodejs\node.EXE
    npm: 7.5.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 88.0.4324.182
    Edge: Spartan (44.19041.423.0), Chromium (88.0.705.63)
    Internet Explorer: 11.0.19041.1
  npmPackages:
    @sveltejs/adapter-netlify: 1.0.0-next.1
    @sveltejs/adapter-vercel: 1.0.0-next.2
    @sveltejs/kit: 1.0.0-next.34
    @sveltejs/snowpack-config: next => 1.0.0-next.3
    svelte: ^3.29.0 => 3.32.3
```

**Severity**
This makes the Netlify and Vercel adapters unusable. The user might be able to postprocess the adapter results to CJS themselves, but practically speaking it's a blocker for those platforms.

**Additional context**
In an ideal world, the lambdas could run ESM, but it seems the adapter output might need to be translated to CJS instead.

This would likely affect all variants of AWS lambdas. AWS [recently added Node 14 support](https://aws.amazon.com/blogs/compute/node-js-14-x-runtime-now-available-in-aws-lambda/) but it does not mention supporting ESM imports, so I'm not sure of their stance on it. There's also an [open Netlify forum post](https://community.netlify.com/t/lambda-function-es-module-support/30673/5) about it.

#349 is likely related, as this impedes the goal to have everything in ESM.

**Comment by GrygrFlzr** [comment]
Closing this as the relevant PRs have been merged. Further discussion should go to #503 instead.

**Comment by spences10** [comment]
> This issue can be closed as soon as #483, #485, and #486 all get merged.
> 
> * Running Vercel example: https://vercel-test-grygrflzr.vercel.app/
> * Running Netlify example: https://kit-test.netlify.app/

Do you have a code example for the Vercel example you have here @GrygrFlzr, please?
